### PR TITLE
Makes the type definitions compatible with WebdriverIO 4.4.0.

### PR DESCRIPTION
**Pull request by fredrik-smedberg**
Please fill in this template.

- [X] Prefer to make your PR against the `types-2.0` branch.
- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code.
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped#common-mistakes).
- [/] Run `npm run lint -- package-name` if a `tslint.json` is present.

If adding a new definition:
- [ ] The package does not provide its own types, and you can not add them.
- [ ] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [ ] Run `tsc` without errors.
- [ ] Include the required [files](https://github.com/DefinitelyTyped/DefinitelyTyped#create-a-new-package) and header. Base these on the README, *not* on an existing project.

If changing an existing definition:
- [X] Provide a URL to documentation or source code which provides context for the suggested changes: http://webdriver.io
- [X] Increase the version number in the header if appropriate.

Makes the type definitions compatible with WebdriverIO 4.4.0. Merged my changes (removed all callbacks) with WillLuce pull request #12416.

### CODE REVIEW

**Comment by andy-ms** [comment]
Please rebase onto the latest `types-2.0` to fix tests. Fixed by #13234.

**Comment by fsmedberg-tc** [comment]
@andy-ms I think I've managed to rebase my fork with all latest changes. Should I do a new pull request now or is this handled automatically? (this is all very new to me)

**Comment by andy-ms** [comment]
The test failure is legitimate now: `webdriverio-tests.ts(64,9): error TS2346: Supplied parameters do not match any signature of call target.` Try running `tsc` locally to see what's the problem.

File: webdriverio/index.d.ts [hunk 1]
```diff
@@ -30,96 +30,116 @@ declare namespace WebdriverIO {
...
+        leftClick<P>(selector: string): Client<P>;
+        leftClick<P>(): Client<P>;
 
         middleClick(selector: string): Client<void>;
-        middleClick<P>(
-            selector: string,
-            callback: (err: any) => P
-        ): Client<P>;
+        middleClick(): Client<void>;
+        middleClick<P>(selector: string): Client<P>;
```
Comment (andy-ms) [inline webdriverio/index.d.ts]: Could just use `selector?: string` rather than these two overloads.

File: webdriverio/index.d.ts [hunk 2]
```diff
@@ -134,17 +154,17 @@ declare namespace WebdriverIO {
...
         ): Client<void>;
+        setValue<P>(values: number | string | Array<string>): Client<void>;
 
         submitForm(selector: string): Client<void>;
-        submitForm<P>(
-            selector: string,
-            callback: (err: any) => P
-        ): Client<void>;
+        submitForm(): Client<void>;
+        submitForm<P>(selector: string): Client<void>;
```
Comment (andy-ms) [inline webdriverio/index.d.ts]: The type parameter is unused.

**Comment by fsmedberg-tc** [comment]
Is my pull request OK now to merge with the types-2.0 branch?

**Comment by andy-ms** [comment]
See my 2 review comments above.

File: webdriverio/index.d.ts (hunk 1 above)
Comment (fsmedberg-tc) [inline webdriverio/index.d.ts]: You're absolutely right. I was so caught up in search/replace that I forgot to think about the sanity of doing as I did. selector? would be much cleaner. I'll fix it right away.

File: webdriverio/index.d.ts (hunk 2 above)
Comment (fsmedberg-tc) [inline webdriverio/index.d.ts]: Fixed!

**Comment by WillLuce** [comment]
Maybe I'm missing something, but I just tried to update my @types file with this merge, but it doesn't seem to be there.

**Comment by andy-ms** [comment]
That's my fault. Fix is in #13362.

**Comment by christian-bromann** [comment]
@fsmedberg-tc @WillLuce thanks for keeping the type definitions lit 🔥

**Comment by WillLuce** [comment]
@andy-ms I saw that #13362 was merged in, but this still doesn't seem to be pulling in this change. Is there some back end validation that I'm not seeing?

I'm doing this by `npm install --save-dev @types/webdriverio`

**Comment by pmoleri** [comment]
Hi @WillLuce,
This PR reverted changes made in my previous PR #12994.
Is there any reason for that? Am I missing something?

**Comment by grovesNL** [comment]
Slightly unrelated, but have you all been using this definition with promises? I can't tell whether methods like `getTitle` return a promise or not by the signature, which makes it hard to use without knowing each method in advance. I know that the method chaining complicates the signature but I wonder if this could be improved.

**Comment by pmoleri** [comment]
Hi @grovesNL, I've been using them with `await`.
AFAIK, if it returns a `Client<P>` then you need to _resolve_ the promise to get the value, i.e. use `.then()`, `await` or something like `chai-as-promised`.
e.g.:
```
const initialLocation = await app.client.getLocation(selector);
```
`getLocation` returns a `Client<Location>`, so I use await to access the result.

**Comment by grovesNL** [comment]
@pmoleri: thanks! I've been using `await` as you mentioned but the issue is that I can't tell by the return type of `Client<P>` whether it's asynchronous or not.

I'd expect the return type to be some variant of `Promise<P>` instead, but I'm not sure which async methods are chainable as well. It seems like the majority are both async and chainable.

**Pull request merged**